Jellyfin.Api/Controllers/TrickplayController.cs

```diff
@@ -0,0 +1,178 @@
+    /// <param name="trickplayManager">Instance of <see cref="ITrickplayManager"/>.</param>
+    public TrickplayController(
+        ILogger<TrickplayController> logger,
+        IHttpContextAccessor httpContextAccessor,
+        ILibraryManager libraryManager,
+        ITrickplayManager trickplayManager)
+    {
+        _logger = logger;
+        _httpContextAccessor = httpContextAccessor;
+        _libraryManager = libraryManager;
+        _trickplayManager = trickplayManager;
+    }
+
+    /// <summary>
+    /// Gets an image tiles playlist for trickplay.
+    /// </summary>
+    /// <param name="itemId">The item id.</param>
+    /// <param name="width">The width of a single tile.</param>
+    /// <param name="mediaSourceId">The media version id, if using an alternate version.</param>
+    /// <response code="200">Tiles stream returned.</response>
+    /// <returns>A <see cref="FileResult"/> containing the trickplay tiles file.</returns>
+    [HttpGet("Videos/{itemId}/Trickplay/{width}/tiles.m3u8")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesPlaylistFile]
+    public ActionResult GetTrickplayHlsPlaylist(
+        [FromRoute, Required] Guid itemId,
+        [FromRoute, Required] int width,
+        [FromQuery] string? mediaSourceId)
+    {
+        return GetTrickplayPlaylistInternal(width, mediaSourceId ?? itemId.ToString("N"));
+    }
+
+    /// <summary>
+    /// Gets a trickplay tile grid image.
+    /// </summary>
+    /// <param name="itemId">The item id.</param>
+    /// <param name="width">The width of a single tile.</param>
+    /// <param name="index">The index of the desired tile grid.</param>
+    /// <param name="mediaSourceId">The media version id, if using an alternate version.</param>
+    /// <response code="200">Tiles image returned.</response>
+    /// <response code="200">Tiles image not found at specified index.</response>
+    /// <returns>A <see cref="FileResult"/> containing the trickplay tiles image.</returns>
+    [HttpGet("Videos/{itemId}/Trickplay/{width}/{index}.jpg")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesImageFile]
+    public ActionResult GetTrickplayHlsPlaylist(
+        [FromRoute, Required] Guid itemId,
+        [FromRoute, Required] int width,
+        [FromRoute, Required] int index,
+        [FromQuery] string? mediaSourceId)
+    {
+        var item = _libraryManager.GetItemById(mediaSourceId ?? itemId.ToString("N"));
+        if (item is null)
+        {
+            return NotFound();
+        }
+
+        var path = _trickplayManager.GetTrickplayTilePath(item, width, index);
+        if (System.IO.File.Exists(path))
+        {
+            return PhysicalFile(path, MediaTypeNames.Image.Jpeg);
+        }
+
+        return NotFound();
+    }
+
+    private ActionResult GetTrickplayPlaylistInternal(int width, string mediaSourceId)
```

Comment:
Maybe it would be a good idea to put that into its own service?

Reply:
I wrote the playlist logic as part of the controller to mimic the current way external playlists seem to be structured, like the SubtitleController:
https://github.com/jellyfin/jellyfin/blob/bb5bf0277ae33fd4668e00bd2f1a85e89aef466a/Jellyfin.Api/Controllers/SubtitleController.cs#L324-L329

I'm not sure if the code for generating playlists does or doesn't have reusability outside of HLS, but if we did want it to be generated in a publicly accessible way the [DynamicHlsPlaylistGenerator](https://github.com/jellyfin/jellyfin/blob/master/src/Jellyfin.MediaEncoding.Hls/Playlist/DynamicHlsPlaylistGenerator.cs) is the only place I can think to put it.

Reply:
I am not so much interested in reusability but more in another extension point for a plugin the adapt or rewrite the playlist generation. That would make a good point to join another playlist provider into.

Reply:
Extending `DynamicHlsPlaylistGenerator` might be the right place as an alternative to its own service though.

Reply:
It looks like trickplay might be outside the scope of ```DynamicHlsPlaylistGenerator``` given it's description: "Generator for dynamic HLS playlists where the segment lengths aren't known in advance." Given the relative scatter of Hls/Streaming code (e.g. DynamicHlsPlaylistGenerator, DynamicHlsHelper, StreamingHelpers, HlsHelpers, the different playlist controllers) and having just started working with the codebase, I'm going to stick with the ```SubtitleController```-type logic for now as it seems to be the closest analog.

If/when there is a general consensus on where specifically the playlist code should be moved or if it should be further abstracted altogether, it shouldn't be difficult to accommodate that.

Reply:
Non-Action methods in controllers are possible but also not the cleanest thing (separation of concerns). I would still say an service for that kind of logic would be best as such business logic is (mostly) located inside services. I see not harm in creating a new interface and implementation alongside the `ITrickplayManager` like a `ITrickplayPlaylistGenerator`

Reply:
I think this function can live inside of TrickPlayManager? I agree that it doesn't belong in the controller

Reply:
> I think this function can live inside of TrickPlayManager? I agree that it doesn't belong in the controller

It's difficult to implement in the manager in a way that makes sense. The trickplay playlist is intrinsically linked to the trickplay controller URL that serves the images, as well as the current http context to include the api key in the playlist.

The only way to make a playlist then would be to pass in the api key, which does make it acessible to other api controllers, but doesn't seem like a realistic use case. 

I could remove the authentication entirely, or implement it as stated above, but given that the current implementation is consistent with other HLS endpoints like the SubtitleController, I don't see a very compelling reason to do so. https://github.com/jellyfin/jellyfin/blob/bb5bf0277ae33fd4668e00bd2f1a85e89aef466a/Jellyfin.Api/Controllers/SubtitleController.cs#L324-L329